### PR TITLE
Like and dislike states added

### PR DESCRIPTION
**Pull request by javierbustillo**

### CODE REVIEW

File: DAO/MessageDAO.py [hunk 1]
```diff
@@ -161,3 +161,19 @@ def insert_reply(self, message, uid, mid, cid, img=None):
         self.conn.commit()
         return reply_id
 
+    def remove_vote(self, mid, uid, upvote):
```
Comment (ChristianPerez34) [inline DAO/MessageDAO.py:164]: Is this like in facebook? I mean like when you like a post but then hit the like button again to remove the like?

File: DAO/MessageDAO.py [hunk 2]
```diff
@@ -161,3 +161,19 @@ def insert_reply(self, message, uid, mid, cid, img=None):
...
+        cursor = self.get_cursor()
+        query = "SELECT * FROM vote WHERE mid = %s and uid = %s"
+        cursor.execute(query, (mid, uid))
+        vote = [vote for vote in cursor.fetchall()]
```
Comment (ChristianPerez34) [inline DAO/MessageDAO.py:168]: I'm assuming that there can only be one vote per user. In that case you can do the following

```python
delete = cursor.fetchone()['upvote'] == upvote
```

File: Handlers/Message.py [hunk 3]
```diff
@@ -108,3 +104,17 @@ def get_trending_hashtags(self):
...
 
+    def _vote_message(self, mid, upvote):
+        user = get_jwt_identity()
+        uid = UserDAO().get_user_by_username(user)['uid']
+        try:
+            self.dao.vote_message(mid, uid, upvote)
+            msg = 'Successfully added vote'
+        except:
+            self.dao.conn.rollback()
+            deleted = self.dao.remove_vote(mid, uid, upvote)
```
Comment (ChristianPerez34) [inline Handlers/Message.py:115]: If there is a rollback how come you have to delete the vote? Doesn't rollback handle this?

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

File: DAO/MessageDAO.py (hunk 1 above)
Comment (javierbustillo) [reply DAO/MessageDAO.py:164]: Yeah

File: DAO/MessageDAO.py (hunk 2 above)
Comment (javierbustillo) [reply DAO/MessageDAO.py:168]: Will the way I did it affect functionality?

File: Handlers/Message.py (hunk 3 above)
Comment (javierbustillo) [reply Handlers/Message.py:115]: Esto corre si ya existe un voto, en ese caso hay un databaseerror and the transaction is aborted. Tienes que hacer un rollback para volver la base de dato a un estado stable por si quieres hacer otra operacion

File: DAO/MessageDAO.py (hunk 2 above)
Comment (ChristianPerez34) [reply DAO/MessageDAO.py:168]: Noted

**Review by ChristianPerez34 — APPROVED** [review]

**Pull request merged**